pymongo/pyopenssl_context.py

```diff
@@ -198,66 +209,77 @@ def __init__(self, protocol):
         self._ctx.set_ocsp_client_callback(callback=_ocsp_callback, data=self._callback_data)
 
     @property
-    def protocol(self):
+    def protocol(self) -> int:
         """The protocol version chosen when constructing the context.
         This attribute is read-only.
         """
         return self._protocol
 
-    def __get_verify_mode(self):
+    def __get_verify_mode(self) -> VerifyMode:
         """Whether to try to verify other peers' certificates and how to
         behave if verification fails. This attribute must be one of
         ssl.CERT_NONE, ssl.CERT_OPTIONAL or ssl.CERT_REQUIRED.
         """
         return _REVERSE_VERIFY_MAP[self._ctx.get_verify_mode()]
 
-    def __set_verify_mode(self, value):
+    def __set_verify_mode(self, value: VerifyMode) -> None:
         """Setter for verify_mode."""
 
-        def _cb(connobj, x509obj, errnum, errdepth, retcode):
+        def _cb(
+            connobj: _SSL.Connection,
+            x509obj: _crypto.X509,
+            errnum: int,
+            errdepth: int,
+            retcode: int,
+        ) -> bool:
             # It seems we don't need to do anything here. Twisted doesn't,
             # and OpenSSL's SSL_CTX_set_verify let's you pass NULL
             # for the callback option. It's weird that PyOpenSSL requires
             # this.
-            return retcode
+            return bool(retcode)
```

Comment:
Can you add a comment that this is now optional in pyopenssl >=20 and can be removed when we bump our min supported version? 

> Make verification callback optional in Context.set_verify. If omitted, OpenSSL’s default verification is used. [#933](https://github.com/pyca/pyopenssl/pull/933)

See: https://www.pyopenssl.org/en/latest/changelog.html#id47

Reply:
yes, done